dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/ErrorContext.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.annotations.processor;
+
+import com.google.errorprone.annotations.CompileTimeConstant;
+import com.palantir.logsafe.Arg;
+import javax.lang.model.element.Element;
+
+public interface ErrorContext {
+    void reportError(@CompileTimeConstant String message, Element element, Arg<?>... args);
+
+    void reportError(@CompileTimeConstant String message, Element element, Throwable throwable);
```

Comment:
Should this also include `Arg<?>... args`? Suppose it can be updated once args are needed. Probably not important to use safe-logging in this processor, but I think it's worthwhile for consistency.

Reply:
I'm just adding and fixing this up as I go along; I can make a final pass once everything is finished to clean up things like this. I think I'll have some better things in here once I've done more in-depth validation.